api/src/main/java/org/apache/iceberg/IncrementalAppendScan.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+package org.apache.iceberg;
+
+/**
+ * API for configuring an incremental table scan for appends only snapshots
+ */
+public interface IncrementalAppendScan extends Scan<IncrementalAppendScan> {
```

Comment:
I think this can be abstracted into `IncrementalScan`, because not only append-only, changelog incremental scan also needs to specify `from` and `to` snapshots. We can define the `IncrementalScan` interface, and then return different concrete implementations in `Table` through different methods, for example:

````
public interface IncrementalScan extends Scan<IncrementalScan> {...}

abstract class BaseIncrementalScan implements IncrementalScan {...}
public class AppendOnlyIncrementalScan extends BaseIncrementalScan {...}
public class ChangelogIncrementalScan extends BaseIncrementalScan {...}

public interface Table {
   ...
  IncrementalScan newAppendIncrementalScan();
  IncrementalScan newChangelogIncrementalScan();
  ...
}
````

Reply:
@Reo-LEI conceptually I agree with you. @rdblue prefer to do the refactoring when we come to the changelog incremental scan. 

The reason we didn't add `BaseIncrementalScan` is that right now there is no difference between `BaseIncrementalScan` and `IncrementalAppendScan`. In the future, we can extract `BaseIncrementalScan` out or we can have `IncrementalChangelogScan` extending `IncrementalAppendScan`. Personally, I also prefer the `BaseIncrementalScan`.